cumulus/pallets/parachain-system/src/validate_block/implementation.rs

```diff
@@ -301,11 +305,33 @@ where
 					);
 				}
 			},
-		)
+		);
+
+		if block_index + 1 != num_blocks {
+			let mut changes = overlay
+				.drain_storage_changes(
+					&backend,
+					<PSC as frame_system::Config>::Version::get().state_version(),
+				)
+				.expect("Failed to get drain storage changes from the overlay.");
+
+			drop(backend);
+
+			// We just forward the changes directly to our db.
+			changes.transaction.drain().into_iter().for_each(|(_, (value, count))| {
+				// We only care about inserts and not deletes.
+				if count > 0 {
+					db.insert(EMPTY_PREFIX, &value);
```

Comment:
What is up with this `EMPTY_PREFIX` here? I see that internally, the value is hashed for the key. Makes sense, but when would we need the  prefix here?